src/paths/steiner.cpp

```diff
@@ -0,0 +1,251 @@
+#include "igraph_paths.h"
+#include "igraph_adjlist.h"
+#include "igraph_types.h"
+#include "igraph_memory.h"
+#include "igraph_error.h"
+#include "igraph.h"
+
+#include <string.h>
+#include <math.h>
+#include <map>
+#include <limits.h>
+#include <vector>
+#include <set>
+
+std::map<std::set<igraph_integer_t>, igraph_integer_t> subsetMap;
+
+void printSubsets(std::set<std::set<igraph_integer_t>> allSubsets)
+{
+	printf("Subsets :\n{\n");
+	for (auto i = allSubsets.begin() ; i != allSubsets.end() ; ++i )
+	{
+		printf("\t{ ");
+		for (auto j = (*i).begin() ; j != (*i).end() ; j++)
+		{
+			if (j != (*i).begin()) { printf(", ");}
+			printf("%ld",*j);
```

Comment:
This is causing a compilation error on macOS. The type of `*j`, which is `igraph_integer_t` is not necessarily the same as `long int`. Use `"%" IGRAPH_PRId` instead of `"%ld"`.

That said, all `printf`s will need to be removed (or hidden behind debug macros) before the PR can be merged. igraph is used from R, and CRAN policy is not to allow `printf` in the code.